kbs/src/plugins/implementations/resource/id_key.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2025 by Red Hat.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{ResourceDesc, StorageBackend};
+use aes_gcm::{aead::Aead, Aes256Gcm, Key, KeyInit};
+use anyhow::{anyhow, Context, Result};
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use generic_array::GenericArray;
+use openssl::{
+    pkey::Private,
+    rsa::{Padding, Rsa},
+};
+use p384::{ecdh, PublicKey, SecretKey};
+use serde::{Deserialize, Serialize};
+use sha2::Sha256;
+
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+pub struct IdKeyConfig;
+
+impl Default for IdKeyConfig {
+    fn default() -> Self {
+        Self
+    }
+}
+
+pub struct IdKeyBackend {
+    idkey_non_hsm: Rsa<Private>,
+    ecdh_secret: SecretKey,
+}
+
+#[async_trait::async_trait]
+impl StorageBackend for IdKeyBackend {
+    async fn read_secret_resource(
+        &self,
+        resource_desc: ResourceDesc,
+        data: &[u8],
+    ) -> Result<Vec<u8>> {
+        if resource_desc.repository_name != *"id-key" {
+            return Err(anyhow!("invalid repository name"));
+        }
+
+        if resource_desc.resource_type == *"ecdh-pub-sec1" {
+            return Ok(self.ecdh_pubkey_sec1());
+        }
+
+        if resource_desc.resource_type != *"nek" {
+            return Err(anyhow!("invalid resource type"));
+        }
+
+        let ecdh: IdKeyEcdh =
+            serde_json::from_slice(data).context("unable to deserialize request body")?;
```

Comment:
I'm not sure this is going to work.

When making a resource request, the TEE public key is not in the request body. It's in the attestation token. Check out how we extract the public key [here](https://github.com/confidential-containers/trustee/blob/main/kbs/src/api_server.rs#L244) when we use it to encrypt the response.

Reply:
This isn't using the TEE public key. This is because I'm only supporting ECDH, and not leaving open the use of RSA like JWK's allow. Would you prefer it use JWK?

Reply:
So you're expecting the request to have an additional key in it? That seems like an extension of the protocol.

Reply:
I was under impression the protocol was a bit flexible with this. I'm using the protocol in an unconventional way anyway w/r/t needing to first fetch the server's ECDH public key before actually requesting the resource.

Reply:
There are no laws against it, but it's not totally ideal. This would fit better in another plugin. We do support query strings in the request. Any chance you could put the public key in there?

Reply:
Yea, that could be done. I'd actually probably prefer that too, as it basically removes the need for the `kbs-types` PR I previously mentioned.